File: leetcode/C++/intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> sol;
        unordered_map<int,int> um;
        const vector<int> &nums = (nums1.size()>nums2.size())? nums2 : nums1;
        const vector<int> &comps = (nums1.size()>nums2.size())? nums1 : nums2;
        for(const int &n : nums) ++um[n];
        for(const int &n : comps) if(um[n]-->0) sol.push_back(n); 
        return sol;
    }
};

File: leetcode/Python/intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        sol, um = [], defaultdict(int)
        nums = nums1 if len(nums1) > len(nums2) else nums2
        comps = nums2 if len(nums1) > len(nums2) else nums1
        for n in nums:
            um[n]+=1
        for n in comps:
            if um[n]>0:
                um[n]-=1
                sol.append(n)
        return sol